File: scripts/parity/scenarios/py/helix.py
import fullcontrol as fc
from math import tau

# Create a helix using helixZ
centre_point = fc.Point(x=50, y=50, z=0)
start_radius = 10
end_radius = 10
start_angle = 0
n_turns = 5
pitch_z = 0.4
segments = 320
clockwise = True

steps = fc.helixZ(centre_point, start_radius, end_radius, start_angle, n_turns, pitch_z, segments, clockwise)

res = fc.transform(steps, result_type='gcode')
print(res)

File: scripts/parity/scenarios/js/helix.mjs
// Parity scenario: helix using helixZ geometry function
import { Point, helixZ, transform } from '../../../../dist/index.js'

// Create a helix using helixZ
const centre_point = new Point({ x: 50, y: 50, z: 0 })
const start_radius = 10
const end_radius = 10
const start_angle = 0
const n_turns = 5
const pitch_z = 0.4
const segments = 320
const clockwise = true

const steps = helixZ(centre_point, start_radius, end_radius, start_angle, n_turns, pitch_z, segments, clockwise)

const g = transform(steps, 'gcode', { show_tips: false, show_banner: false }).gcode
process.stdout.write(g.trimEnd() + '\n')
